packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -314,6 +316,67 @@ function completeWork(
 ): Fiber | null {
   const newProps = workInProgress.pendingProps;
 
+  // Reset the child expiration time by bubbling it from the children. Skip
+  // if the children are hidden and we're not rendering at hidden priority.
```

Comment:
There are a number of little fixes and refactoring in this commit that isn't directly related to the hiding of children. Such as moving this section and the extra arguments and move of retrySuspendedRoot.

It would be nice to have a separate PR for those since they seem like useful clean up regardless.